rss2email/email.py

```diff
@@ -146,8 +146,11 @@ def get_message(sender, recipient, subject, body, content_type,
         message.set_payload(body, charset=charset)
     if extra_headers:
         for key,value in extra_headers.items():
-            encoding = guess_encoding(value, encodings)
-            message[key] = _Header(value, encoding)
+            if config.getboolean(section, 'encode-date-field') or key != 'Date':
+                encoding = guess_encoding(value, encodings)
```

Comment:
what exactly does “encoding the date field” mean? Can you give some more background?

Reply:
Also a FAQ entry would be much appreciated, this sounds like an issue other users might stumble into.

Reply:
The _Header function QP (RFC 2047) encodes the header contents, and it does so for all headers, including Date.

Date: Mon, 30 Mar 2020 09:18:24 -0000
becomes
Date: =?utf-8?q?Mon=2C_30_Mar_2020_09=3A18=3A24_-0000?=

A locale might generate non-ASCII characters for the day, month, or timezone, and must be QP encoded, or risk being rejected by smtp servers. The Thunderbird issue questioned if encoding the date field violated both RFC 822 and the newer RFC 2822 which defines a syntax that explicitly uses english names for day, month and timezone. Any strict MUA would reject QP encoded date headers due the sender's violation of the RFC.

Another option is to force english in srtftime().

```
with setlocale('C'):
    return _time.strftime("%a, %d %b %Y %H:%M:%S -0000", datetime)
```

Then r2e wouldn't be violating the RFC and the encode-date-field option wouldn't be needed. The 'Date' field check to bypass _Header() would still be needed. I am still investigating why, but it would be nice if QP encoding wasn't always enabled for all headers. Printing _Header() output doesn't QP encode it, even if it conains utf-8 characters, something later on is forcing it to always encode. If that was fixed, it would not only reduce the size of outgoing emails but checking for 'Date' field wouldn't be needed.

Reply:
If we can solve it generally without undue backwards incompatible changes, I’d be in favor of that. If that’s not viable, adding an option and an troubleshooting entry is a good next-best solution.